scripts/maps_score.py

```diff
@@ -0,0 +1,198 @@
+"""Compute MAPS score"""
+
+from typing import List, Union
+import hail as hl
+import sys
+
+
+def downsampling_counts_expr(
+    ht: Union[hl.Table, hl.MatrixTable],
+    pop: str = 'global',
+    variant_quality: str = 'adj',
+    singleton: bool = False,
+    impose_high_af_cutoff: bool = False,
+) -> hl.expr.ArrayExpression:
+    indices = hl.zip_with_index(ht.freq_meta).filter(
+        lambda f: (f[1].size() == 3)
+        & (f[1].get('group') == variant_quality)
+        & (f[1].get('pop') == pop)
+        & f[1].contains('downsampling')
+    )
+    sorted_indices = hl.sorted(indices, key=lambda f: hl.int(f[1]['downsampling'])).map(
+        lambda x: x[0]
+    )
+
+    # TODO: this likely needs to be fixed for aggregations that return missing
+
+    def get_criteria(i):
+        if singleton:
+            return hl.int(ht.freq[i].AC == 1)
+        elif impose_high_af_cutoff:
+            return hl.int((ht.freq[i].AC > 0) & (ht.freq[i].AF <= 0.001))
+        else:
+            return hl.int(ht.freq[i].AC > 0)
+
+    return hl.agg.array_sum(hl.map(get_criteria, sorted_indices))
+
+
+def count_variants(
+    ht: hl.Table,
+    count_singletons: bool = False,
+    count_downsamplings=(),
+    additional_grouping=(),
+    partition_hint: int = 100,
+    omit_methylation: bool = False,
+    return_type_only: bool = False,
+    force_grouping: bool = False,
+    singleton_expression: hl.expr.BooleanExpression = None,
+    impose_high_af_cutoff_here: bool = False,
+):
+    """
+    Count variants by context, ref, alt, methylation_level
```

Comment:
The readability would improve if you add some detail on each function parameter and the return value. Something like here: https://github.com/populationgenomics/cpg-qc/blob/main/cpg_qc/hard_filtering.py#L28-L50

Also, don't forget to add type hints to the return value as well (e.g. `-> hl.Table`)

Reply:
This function is from https://github.com/macarthur-lab/gnomad_lof, and I just copied it here to make sure the pipeline can run by itself. I forgot to add links to where I copied the function(s) from, so now they are included in the docstring.

Reply:
I think one way would be to add gnomad_lof as a [git submodule](https://git-scm.com/book/en/v2/Git-Tools-Submodules):

```
git submodule add https://github.com/macarthur-lab/gnomad_lof
```

Then add it to the PYTHONPATH internally, and import them as python modules in `maps_score.py`. For `hailctl dataproc` submission, you'd have to append the gnomad_lof files that you are using into the `--pyfiles` command as well.

Though I don't see anything criminal in copying after you added the pointers to the original :)

Reply:
Submodules are actually a pretty good solution to my problem, didn't think about them! I might consider them in a separate PR